packages/sats-wagmi/src/hooks/useBalance.tsx

```diff
@@ -29,10 +30,31 @@ const useBalance = (props: UseBalanceProps = {}) => {
       }
 
       const esploraClient = new EsploraClient(network);
-
-      const { confirmed, unconfirmed, total } = await esploraClient.getBalance(address);
-
-      return { confirmed: BigInt(confirmed), unconfirmed: BigInt(unconfirmed), total: BigInt(total) };
+      const ordinalsClient = new OrdinalsClient(network);
+
+      const addressInfo = getAddressInfo(address);
+
+      if (addressInfo.type === AddressType.p2tr) {
+        // cardinal = return UTXOs not containing inscriptions or runes
+        const outputsFromAddress = await ordinalsClient.getOutputsFromAddress(address, 'cardinal');
+        const taprootBalance = outputsFromAddress.reduce((acc, cur) => {
+          return acc + cur.value;
+        }, 0);
+
+        return {
+          confirmed: BigInt(taprootBalance),
+          unconfirmed: BigInt(0),
+          total: BigInt(taprootBalance)
+        };
```

Comment:
another option to calculate balance for taproot address could be calling
- esploraClient.getBalance(address);
- ordinalsClient.getOutputsFromAddress(address, 'runic')
- ordinalsClient.getOutputsFromAddress(address, 'inscribed')

2 calls to ordinals client will returns us all UTXOs that we do not want to spend from the address. Then we will subtract that value from `confirmed` and `total`.